astropy/units/tests/test_quantity_annotations.py

```diff
@@ -2,9 +2,107 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 import pytest
-
 from astropy import units as u  # pylint: disable=W0611
 
+import typing as T
+from astropy.units._typing import Annotated
+
+
+def test_type_annotations():
+    # Basic Annotation
```

Comment:
I was confused with this test here. Is it to check that we ignore an annotation that is not useful for quantity input? Maybe clarify in comment.

Actually, I see it is used below, in `test_quantity_typing`, so remove here?

Reply:
> ignore an annotation that is not useful for quantity input?

Yeah, I adapted that test from an existing one.
The tests in ``test_quantity_typing`` only test that Quantity[] works and can be used with ``typing`` objects like Optional and Union, it does not test the decorator. I think both tests are needed.